ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java

```diff
@@ -284,7 +284,7 @@ public final int doPhase2Abort ()
 
 		org.omg.CosTransactions.Status s = get_status();
 
-		if ((s == org.omg.CosTransactions.Status.StatusCommitted)
+		if ((s == org.omg.CosTransactions.Status.StatusCommitted && preparedList.size() == 0)
```

Comment:
For more background I designed this to match the behavior in JTA now: https://github.com/jbosstm/narayana/blob/master/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/subordinate/TransactionImple.java#L219 however I still wanted to allow the existing massage for cases where it was still safe.